core/component/auth_cache.go

```diff
@@ -0,0 +1,48 @@
+// Copyright © 2016 The Things Network
+// Use of this source code is governed by the MIT license that can be found in the LICENSE file.
+
+package component
+
+import (
+	"encoding/json"
+	"time"
+
+	"github.com/TheThingsNetwork/go-account-lib/cache"
+	"golang.org/x/oauth2"
+)
+
+func authCacheKey(appID, accessKey string) string {
+	return appID + ":" + accessKey
```

Comment:
Consider hashing the access key as it has an arbitrary length. Note that you will need to validate against collisions then however.